core/src/main/scala/cats/Parallel.scala

```diff
@@ -0,0 +1,220 @@
+package cats
+
+import cats.arrow.FunctionK
+
+/**
+  * Some types that form a FlatMap, are also capable of forming an Apply that supports parallel composition.
+  * The NonEmptyParallel type class allows us to represent this relationship.
+  */
+trait NonEmptyParallel[M[_], F[_]] extends Serializable {
+  /**
+    * The Apply instance for F[_]
+    */
+  def applicative: Apply[F]
+
+  /**
+    * The FlatMap instance for M[_]
+    */
+  def monad: FlatMap[M]
```

Comment:
could we call this `def flatMap: FlatMap[M]` or something? I don't like returning `FlatMap` with a `monad`.